Sources/ArgumentParser/Usage/DumpHelpInfoGenerator.swift

```diff
@@ -0,0 +1,186 @@
+//===----------------------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Argument Parser open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+@_implementationOnly import Foundation
+
+internal struct DumpHelpInfoGenerator {
+  struct CommandInfo: Codable {
```

Comment:
I think `CommandInfo` and `ArgumentInfo` will probably be independently useful — can you move them out to be top-level types?

Reply:
Addressed at 94a81ed